fsspec/spec.py

```diff
@@ -848,25 +848,26 @@ def get_file(
         """Copy single remote file to local"""
         if isfilelike(lpath):
             outfile = lpath
-        else:
-            if self.isdir(rpath):
-                os.makedirs(lpath, exist_ok=True)
-                return None
+        elif self.isdir(rpath):
+            os.makedirs(lpath, exist_ok=True)
+            return None
```

Comment:
These don't seem mutually exclusive, but I've preserved logic that was here before. Not sure what `outfile` argument is even used for, but it is strange that `makedirs`-behaviour (which is also odd by itself) doesn't affect it.

Reply:
Also calling `isdir` on every file is very wasteful. Ideally, we should get rid of this at some point in the future, as it seems to be a misplaced piece of logic from `fs.get` rather than something that is intentional here. From user's perspective I would expect `IsADirectoryError` here.